src/test/miniminer_tests.cpp

```diff
@@ -0,0 +1,477 @@
+            if (target_feerate <= tx5_feerate_unbumped) {
+                BOOST_CHECK_EQUAL(it5_spent, 0);
+            } else {
+                // Difference is fee to bump tx5 from current to target feerate, without tx6.
+                BOOST_CHECK_EQUAL(it5_spent, target_feerate.GetFee(tx5_dimensions.vsize) - tx5_dimensions.mod_fee);
+            }
+        }
+    }
+}
+
+BOOST_FIXTURE_TEST_CASE(miniminer_overlap, TestChain100Setup)
+{
+    CTxMemPool& pool = *Assert(m_node.mempool);
+    LOCK2(::cs_main, pool.cs);
+    TestMemPoolEntryHelper entry;
+
+    const CAmount low_fee{CENT/2000};
+    const CAmount med_fee{CENT/200};
+    const CAmount high_fee{CENT/10};
+
+    // Create 3 parents of different feerates, and 1 child spending from all 3.
+    const auto tx1 = make_tx({COutPoint{m_coinbase_txns[0]->GetHash(), 0}}, /*num_outputs=*/2);
+    pool.addUnchecked(entry.Fee(low_fee).FromTx(tx1));
+    const auto tx2 = make_tx({COutPoint{m_coinbase_txns[1]->GetHash(), 0}}, /*num_outputs=*/2);
+    pool.addUnchecked(entry.Fee(med_fee).FromTx(tx2));
+    const auto tx3 = make_tx({COutPoint{m_coinbase_txns[2]->GetHash(), 0}}, /*num_outputs=*/2);
+    pool.addUnchecked(entry.Fee(high_fee).FromTx(tx3));
+    const auto tx4 = make_tx({COutPoint{tx1->GetHash(), 0}, COutPoint{tx2->GetHash(), 0}, COutPoint{tx3->GetHash(), 0}}, /*num_outputs=*/3);
+    pool.addUnchecked(entry.Fee(high_fee).FromTx(tx4));
+
+    // Create 1 grandparent and 1 parent, then 2 children.
+    const auto tx5 = make_tx({COutPoint{m_coinbase_txns[3]->GetHash(), 0}}, /*num_outputs=*/2);
+    pool.addUnchecked(entry.Fee(high_fee).FromTx(tx5));
+    const auto tx6 = make_tx({COutPoint{tx5->GetHash(), 0}}, /*num_outputs=*/3);
+    pool.addUnchecked(entry.Fee(low_fee).FromTx(tx6));
+    const auto tx7 = make_tx({COutPoint{tx6->GetHash(), 0}}, /*num_outputs=*/2);
+    pool.addUnchecked(entry.Fee(med_fee).FromTx(tx7));
+    const auto tx8 = make_tx({COutPoint{tx6->GetHash(), 1}}, /*num_outputs=*/2);
+    pool.addUnchecked(entry.Fee(high_fee).FromTx(tx8));
+
+    std::vector<CTransactionRef> all_transactions{tx1, tx2, tx3, tx4, tx5, tx6, tx7, tx8};
+    std::vector<int64_t> tx_vsizes;
+    tx_vsizes.reserve(all_transactions.size());
+    for (const auto& tx : all_transactions) tx_vsizes.push_back(GetVirtualTransactionSize(*tx));
+
+    std::vector<COutPoint> all_unspent_outpoints({
+        COutPoint{tx1->GetHash(), 1},
+        COutPoint{tx2->GetHash(), 1},
+        COutPoint{tx3->GetHash(), 1},
+        COutPoint{tx4->GetHash(), 0},
+        COutPoint{tx4->GetHash(), 1},
+        COutPoint{tx4->GetHash(), 2},
+        COutPoint{tx5->GetHash(), 1},
+        COutPoint{tx6->GetHash(), 2},
+        COutPoint{tx7->GetHash(), 0},
+        COutPoint{tx8->GetHash(), 0}
+    });
+    for (const auto& outpoint : all_unspent_outpoints) BOOST_CHECK(!pool.isSpent(outpoint));
+
+    const auto tx3_feerate = CFeeRate(high_fee, tx_vsizes[2]);
```

Comment:
For better readability and maintainability, would be nice to start naming the txs with index zero, to avoid the off-by-one relation (`txN <-> tx_vsizes[N-1]`).

Reply:
Great idea, will do.

Reply:
Implemented in #26152